aten/src/ATen/native/cuda/UpSampleBilinear2d.cu

```diff
@@ -456,6 +457,388 @@ static void upsample_bilinear2d_backward_out_cuda_template(
   });
 }
 
+// Code for upsampling with antialias
+template <typename scalar_t, typename accscalar_t, int interp_size>
+C10_LAUNCH_BOUNDS_1(256) // 256 performs better then 1024
+__global__ void upsample_gen2d_aa_out_frame(
+    const accscalar_t height_scale,
+    const accscalar_t width_scale,
+    const bool align_corners,
+    const PackedTensorAccessor64<scalar_t, 4> idata,
+    PackedTensorAccessor64<scalar_t, 4> odata) {
+
+  const int batchsize = idata.size(0);
+  const int channels = idata.size(1);
+  const int input_height = idata.size(2);
+  const int input_width = idata.size(3);
+  const int output_height = odata.size(2);
+  const int output_width = odata.size(3);
+
+  const int output_x = threadIdx.x + blockIdx.x * blockDim.x;
+  const int output_y = threadIdx.y + blockIdx.y * blockDim.y;
+
+  if (output_x >= output_width || output_y >= output_height) {
+    return;
+  }
+
+  const accscalar_t half = 0.5;
+  const accscalar_t support_h = static_cast<accscalar_t>(
+      (height_scale >= 1.0) ? (interp_size * half) * height_scale : interp_size * half);
+  const accscalar_t support_w = static_cast<accscalar_t>(
+      (width_scale >= 1.0) ? (interp_size * half) * width_scale : interp_size * half);
+
+  const int interp_height = (int)ceilf(support_h) * 2 + 1;
+  const int interp_width = (int)ceilf(support_w) * 2 + 1;
+
+  // Setup weights and a buffer using shared memory
+  extern __shared__ int smem[];
+  scalar_t* wx = reinterpret_cast<scalar_t*>(smem) + interp_width * threadIdx.x;
+  scalar_t* wy = reinterpret_cast<scalar_t*>(smem) + interp_width * blockDim.x + interp_height * threadIdx.y;
+  const int offset = interp_width * blockDim.x + interp_height * blockDim.y;
+  scalar_t *buffer2 = reinterpret_cast<scalar_t*>(smem) + offset + \
+      interp_height * (threadIdx.x + threadIdx.y * blockDim.x);
+
+  // Compute weights and kernel spans
+  int xmin, xsize, ymin, ysize;
+  accscalar_t xcenter, ycenter;
+  upsample_antialias::_compute_weights_span(
+      output_x, input_width, width_scale, support_w, xmin, xsize, xcenter);
+  upsample_antialias::_compute_weights_span(
+      output_y, input_height, height_scale, support_h, ymin, ysize, ycenter);
+
+  typedef scalar_t (*aa_filter_fn_t)(scalar_t);
+  aa_filter_fn_t filter_fn;
+  if (interp_size == 2) {
+    filter_fn = upsample_antialias::bilinear_filter;
+  } else {
+    filter_fn = upsample_antialias::bicubic_filter;
+  }
```

Comment:
@jeffdaily trying to nail down the problem, looks like this issue with this function declaration. Below function `upsample_antialias::_compute_weight` using it returns garbage into `wx`, `wy`. 
Also trying to explicitly call it like `auto o = filter_fn(1.0);` in the code (inside UpSampleBilinear2d.hip) leads to compile crash : 
```
clang-13: error: unable to execute command: Segmentation fault (core dumped)
clang-13: error: clang frontend command failed due to signal (use -v to see invocation)
AMD clang version 13.0.0 (https://github.com/RadeonOpenCompute/llvm-project roc-4.5.0 21422 e2489b0d7ede612d6586c61728db321047833ed8)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /opt/rocm/hip/../llvm/bin
clang-13: note: diagnostic msg: Error generating preprocessed source(s).
CMake Error at torch_hip_generated_UpSampleBilinear2d.hip.o.cmake:200 (message):
  Error generating file
  /pytorch/build/caffe2/CMakeFiles/torch_hip.dir/__/aten/src/ATen/native/hip/./torch_hip_generated_UpSampleBilinear2d.hip.o
```

Reply:
@jeffdaily I restructured the code such that both computations on AMD and NVidia gpus are working correctly now. It was a problem with passing a function pointer into another function.

Reply:
You just beat me to it.  Our compiler does not support passing pointers to functions.  The precedent elsewhere throughout pytorch is to use functors and template parameters.  I just managed to rewrite your PR to do this and I got the tests to pass.

Reply:
Oh, if you do not mind to show me how to code it as elsewhere in pytorch, I'd be happy to change my latest commit.

>  Our compiler does not support passing pointers to device functions.

Good to know about that. I was wondering why it does not complain during the building time ?

Reply:
Not sure why our compiler does not complain during the build, unless you count the time when it crashed on you.

I pushed my changes to my fork so you can look at the approach I took to resolve this.  

https://github.com/pytorch/pytorch/compare/master...jeffdaily:added-interp-antialias-cuda

Specifically this commit: https://github.com/pytorch/pytorch/commit/ff6f67896037f532f75705450b90fd80ab4da9cf

Note that you add a generic template param for the function, pass the functor as a const ref, then call the function as needed.  This pattern works for our compiler.

Similar changes were made recently to onnxruntime.  https://github.com/microsoft/onnxruntime/pull/9466

Reply:
Thanks for the code and the info ! I hope same works with cuda. I'll update your commit to reduce the number of params as interp_size now can be inside the struct.